src/reps/rep-utils.js

```diff
@@ -28,6 +28,77 @@ function escapeNewLines(value) {
   return value.replace(/\r/gm, "\\r").replace(/\n/gm, "\\n");
 }
 
+// Map from character code to the corresponding escape sequence.  \0
+// isn't here because it would require special treatment in some
+// situations.  \b, \f, and \v aren't here because they aren't very
+// common.  \' isn't here because there's no need, we only
+// double-quote strings.
+const escapeMap = {
+  // Tab.
+  9: "\\t",
+  // Newline.
+  0xa: "\\n",
+  // Carriage return.
+  0xd: "\\r",
+  // Quote.
+  0x22: "\\\"",
+  // Backslash.
+  0x5c: "\\\\",
+};
+
+/**
+ * Escape a string so that the result is viewable and valid JS.
+ * In particular, control characters, [fixme]
+ */
+function escapeString(str) {
+  let result = "";
```

Comment:
I wrote a second version using a regular expression and then ran both versions on some largeish strings.

The looping variant performs better when there are many characters to convert.  However, I think this is not likely to be the usual case.

The regexp variant performs better when there are no characters to change.  So, I will do this instead.